### ISSUE TITLE
very unclear how to use transactions with node-sqlite3

### ISSUE THREAD

**Issue by ORESoftware** [TryGhost/node-sqlite3#773]
I am very confused about how to use transactions with this library.
I have multiple Node.js processes that will interact with the db concurrently on a local machine.

In any given process, I need to do:

> 1. read an integer from db
> 2. increment the integer and update db

So I need to do something like:

```js
db.serialize(function(){
  db.run('BEGIN EXCLUSIVE TRANSACTION;');
  db.each('SELECT run_id from suman_run_id', function(err, row){
    db.serialize(function(){
      console.log('row => ', row);
      const val = row.run_id + 1;
      console.log('val => ', val);
      db.run('UPDATE suman_run_id SET run_id = ' + val );
      db.run('COMMIT TRANSACTION;');
      db.close();
    });
  });
});


```

Feel free to correct my code if something is wrong. However my main concern is: This transaction stuff doesn't seem to work if I am trying to run a transaction in another process concurrently. Surely I will get some error "Cannot begin a new transaction as a transaction is already in progress".

I really need some information as to how to use transactions properly across processes with this library.

Please help! This is the simplest use case ever! thanks. If we figure this out I would be happy to update the docs to use this as an example.

**Comment by ORESoftware** [comment]
If I use this code in two different scripts running in parallel:

```js
const async = require('async');

async.forever(function(cb){

  db.serialize(function(){
    db.run('BEGIN EXCLUSIVE TRANSACTION;');
    db.each('SELECT run_id from suman_run_id', function(err, row){
      db.serialize(function(){
        console.log('row => ', row);
        const val = row.run_id + 1;
        console.log('val => ', val);
        db.run('UPDATE suman_run_id SET run_id = ' + val );
        db.run('COMMIT TRANSACTION;',function(err){
          setTimeout(cb, 100);
        });
      });
    });
  });

}, function(err){
  console.error(err.stack || err);
});
```

eventually I will get an error:


```
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: SQLITE_BUSY: database is locked
```

**Comment by ORESoftware** [comment]
maybe this setting?

https://github.com/mapbox/node-sqlite3/wiki/API#databaseconfigureoption-value

**Comment by ORESoftware** [comment]
what I have above seems to work, if I use the EXCLUSIVE condition

**Comment by daniellockyer** [comment]
Closing in favor of https://github.com/TryGhost/node-sqlite3/issues/304
